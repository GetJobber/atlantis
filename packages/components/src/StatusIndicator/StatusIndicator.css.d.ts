declare const styles: {
  readonly "statusIndicator": string;
};
export = styles;

