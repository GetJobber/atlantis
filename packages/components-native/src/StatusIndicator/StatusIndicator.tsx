import React from "react";
import { View } from "react-native";
import { useStyles } from "./StatusIndicator.style";
import { tokens } from "../utils/design";

export type StatusType =
  | "success"
  | "warning"
  | "critical"
  | "inactive"
  | "informative";
interface StatusIndicatorProps {
  readonly status: StatusType;
}

export function StatusIndicator({ status }: StatusIndicatorProps) {
  const styles = useStyles();

  return (
    <View
      testID={`${status}Indicator`}
      style={[
        styles.statusIndicator,
        { backgroundColor: tokens[`color-${status}`] },
      ]}
    />
  );
}
