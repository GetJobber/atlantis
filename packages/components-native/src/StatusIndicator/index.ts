export { StatusIndicator } from "./StatusIndicator";
export type { StatusType, StatusIndicatorProps } from "./StatusIndicator";
