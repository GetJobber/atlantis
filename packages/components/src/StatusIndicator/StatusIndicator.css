:root {
  --statusLabel-icon-diameter: 8px;
}

.statusIndicator {
  display: inline-block;
  width: var(--statusLabel-icon-diameter);
  height: var(--statusLabel-icon-diameter);
  border-radius: 50%;
  background-color: var(--color-success);
  flex-shrink: 0;
}
