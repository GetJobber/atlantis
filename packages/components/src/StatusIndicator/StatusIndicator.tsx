import React from "react";
import styles from "./StatusIndicator.css";

export type StatusType =
  | "success"
  | "warning"
  | "critical"
  | "inactive"
  | "informative";

interface StatusIndicatorProps {
  readonly status: StatusType;
}

export function StatusIndicator({ status }: StatusIndicatorProps) {
  return (
    <div
      style={{ backgroundColor: `var(--color-${status}` }}
      className={styles.statusIndicator}
    />
  );
}
