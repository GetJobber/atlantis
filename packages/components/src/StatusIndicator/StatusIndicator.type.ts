export type StatusIndicatorType =
  | "success"
  | "warning"
  | "critical"
  | "inactive"
  | "informative";
