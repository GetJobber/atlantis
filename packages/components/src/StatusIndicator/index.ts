export { StatusIndicator } from "./StatusIndicator";
export type { StatusIndicatorType } from "./StatusIndicator.type";
