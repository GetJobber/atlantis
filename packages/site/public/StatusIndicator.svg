<svg width="224" height="117" viewBox="0 0 224 117" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect x="75" y="54" width="10" height="10" rx="5" fill="#3F5D69"/>
<rect x="91" y="54" width="10" height="10" rx="5" fill="#388523"/>
<rect x="107" y="54" width="10" height="10" rx="5" fill="#CDB52D"/>
<rect x="123" y="54" width="10" height="10" rx="5" fill="#D24232"/>
<rect x="139" y="54" width="10" height="10" rx="5" fill="#3B9AE8"/>
</svg>
