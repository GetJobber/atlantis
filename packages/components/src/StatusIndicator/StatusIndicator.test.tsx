import { render } from "@testing-library/react";
import React from "react";
import { StatusIndicator } from "./StatusIndicator";

it("renders a success status indicator", () => {
  const { container } = render(<StatusIndicator status="success" />);
  expect(container).toMatchSnapshot();
});

it("renders a warning status indicator", () => {
  const { container } = render(<StatusIndicator status="warning" />);
  expect(container).toMatchSnapshot();
});

it("renders a informative status indicator", () => {
  const { container } = render(<StatusIndicator status="informative" />);
  expect(container).toMatchSnapshot();
});

it("renders a critical status indicator", () => {
  const { container } = render(<StatusIndicator status="critical" />);
  expect(container).toMatchSnapshot();
});

it("renders an inactive status indicator", () => {
  const { container } = render(<StatusIndicator status="inactive" />);
  expect(container).toMatchSnapshot();
});
