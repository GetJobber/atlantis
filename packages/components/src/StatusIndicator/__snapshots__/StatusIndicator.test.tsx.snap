// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a critical status indicator 1`] = `
<div>
  <div
    class="statusIndicator"
  />
</div>
`;

exports[`renders a informative status indicator 1`] = `
<div>
  <div
    class="statusIndicator"
  />
</div>
`;

exports[`renders a success status indicator 1`] = `
<div>
  <div
    class="statusIndicator"
  />
</div>
`;

exports[`renders a warning status indicator 1`] = `
<div>
  <div
    class="statusIndicator"
  />
</div>
`;

exports[`renders an inactive status indicator 1`] = `
<div>
  <div
    class="statusIndicator"
  />
</div>
`;
